dsa/AVL.java

```diff
@@ -0,0 +1,210 @@
+import java.util.Scanner;
+
+public class AVL {
+
+  public static class Node {
+    int data;
+    Node left; // left child
+    Node right; // right child
+
+    int ht = 1; // height
+    int bal = 0; // balance
+
+    Node(int data) {
+      this.data = data;
+    }
+  }
+
+  public static void main(String[] args) {
+
+    try {
+      Scanner sc = new Scanner(System.in);
+      Node root = null;
+
+      while (true) {
+        System.out.println("Your tree: ");
+        display(root);
+
+        System.out.println("Enter 1 to add a node");
+        System.out.println("Enter 2 to remove a node");
+
+        int choice = sc.nextInt();
+
+        if (choice == 1) {
+          System.out.println("Enter the value of the node: ");
+          int val = sc.nextInt();
+          root = add(root, val);
+          System.out.println("-------------------------------------");
+        } else if (choice == 2) {
+          System.out.println("Enter the value of the node: ");
+          int val = sc.nextInt();
+          root = remove(root, val);
+          System.out.println("-------------------------------------");
+        } else {
+          System.out.println("Program terminated successfully.");
+          System.out.println("-------------------------------------");
+          return;
+        }
+      }
+
+    } catch (Exception e) {
+      System.out.println(e);
+    }
+
+  }
+
+  static int height(Node node) {
+    int lh = (node.left == null ? 0 : node.left.ht);  //left child height
```

Comment:
Using abbrevations like lh, rh is not recommended

Reply:
Okay sir.